geode-core/src/main/java/org/apache/geode/internal/metrics/CacheLifecycleMetricsSession.java

```diff
@@ -19,20 +19,26 @@
 
```

Comment:
We also need to protect the call to `addSubregistry()`. I'm not sure about `removeSubregistry()` (I don't know if that ever throws).

Reply:
The User would be invoking `MetricsSession.addRegistry` from `MetricsPublishingService.start(MetricsSession)`. For example, if the User tries to add a `CompositeMeterRegistry` then `CompositeMeterRegistry.add(MeterRegistry)` would throw IllegalArgumentException which our try-catch in `CacheLifecycleMetricsSession.cacheCreated(InternalCache)` would already be catching and logging.

Reply:
On `start(session)`, the service might simply stash the session, and defer adding the subregistry until a later time. In that situation, the `addSubregistry()` call can happen outside the safe context of `start(session)`. The risk is low… Is it low enough?

Reply:
I think the user would have to use their own thread later on to invoke addSubregistry. Throwing to them is probably good enough. If we catch and log it in calls to addSubregistry, then I think we would want to rethrow so they get a Throwable.